File: solutions/2023/9-Mirage_Maintenance/s1.js
const utils = require('../../../templates/javascript/utils.js')
const {submit} = require('../../../aoc-utils/utils.js')
const path = require('path')
const readFileSync = require('fs').readFileSync
const inputPath = path.join(__dirname, 'input')
const data = readFileSync(inputPath, 'utf8')
const lines = data.split(/\n/)

const rows = lines.map(line=>line.split(' ').map(Number))

console.log(rows)


function getDiffRow(row) {
	const diffRow = []
	for (let i=0; i<row.length-1; i++) {
		diffRow.push(row[i+1] - row[i])
	}
	return diffRow
}

function predictNextNum(row) {
	let nextRow = row
	let lastNums = [nextRow[nextRow.length-1]]

	while (!nextRow.every(num=>num===0)) {
		nextRow = getDiffRow(nextRow)
		lastNums.unshift(nextRow[nextRow.length-1])
	}
	
	const nextNum = lastNums.reduce((p,c)=>p+c)
	return nextNum
}

function solution1(rows) {
	let solution1 = 0

	rows.forEach(row => {
		solution1 += predictNextNum(row)
	})

	return solution1
}

const SOLUTION_1 = solution1(rows)

module.exports = {}

// submit(SOLUTION_1)

console.log(`SOLUTION_1: ${SOLUTION_1}`)
// Example solution: 114

File: solutions/2024/23-LAN_Party/s1.py
import os
import sys

current_dir = os.path.dirname(os.path.realpath(__file__))
utils_path = os.path.join(current_dir, "../../../templates/python")
sys.path.append(utils_path)
import utils

input_file = "input_example"
# input_file = "input"

input_path = os.path.join(current_dir, input_file)

with open(input_path, "r", encoding="utf8") as file:
    data = file.read().split("\n")

def solution_1(data):
    print(data)

# def solution_2(data):
#     pass

if __name__ == "__main__":
    solution_1(data)
    # solution_2(data)
td